compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -962,16 +967,67 @@ static void getDefaultMatmulVectorSizes(
   return;
 }
 
+/// Checks if the input and output types of a linalg op are the same, and if so
+/// returns the type. Otherwise, returns failure.
+static FailureOr<Type> nonWideningLinalgElementType(linalg::LinalgOp op) {
+  SmallVector<Type, 3> inputAndOutputElementTypes;
+  for (Value v :
+       llvm::concat<Value>(op.getRegionInputArgs(), op.getRegionOutputArgs())) {
+    inputAndOutputElementTypes.push_back(v.getType());
+  }
+  if (inputAndOutputElementTypes.empty() ||
+      !llvm::all_equal(inputAndOutputElementTypes))
+    return failure();
+  return inputAndOutputElementTypes[0];
+}
+
+/// Utility to compute the tile sizes for AArch64 SME. Unlike other targets, the
+/// tile sizes picked here must exactly match the SME hardware virtual tiles, as
+/// there is currently no support for lowering non-standard shapes.
+static void
+getMatmulAArch64SMEVectorSizes(linalg::LinalgOp op,
+                               SmallVectorImpl<int64_t> &sizes,
+                               SmallVectorImpl<bool> &scalableSizeFlags) {
+  // Double check the operation is one that is supported for lowering to ArmSME.
+  if (!llvm::isa<linalg::MatmulOp, linalg::MatmulTransposeAOp, linalg::FillOp>(
+          op))
+    return;
```

Comment:
Do we need to check `linalg:FillOp`? It is not common to have a single fill op in dispatch. I thought this function is mostly used for matmuls?

Reply:
Nope good point, this only applies to `LinalgContractionOps`, so won't ever see a fill :+1: